lib/Service/DisplayNotifyService.php

```diff
@@ -219,13 +219,15 @@ public function notifyByDisplayGroupId($displayGroupId)
     }
 
     /** @inheritdoc */
-    public function notifyByCampaignId($campaignId)
+    public function notifyByCampaignId($campaignIds)
     {
-        $this->log->debug('Notify by CampaignId ' . $campaignId);
+        foreach ($campaignIds as $campaignId) {
+            $this->log->debug('Notify by CampaignId ' . $campaignId);
 
-        if (in_array('campaign_' . $campaignId, $this->keysProcessed)) {
-            $this->log->debug('Already processed ' . $campaignId . ' skipping this time.');
-            return;
+            if (in_array('campaign_' . $campaignId, $this->keysProcessed)) {
```

Comment:
This means that if any of the campaignIds in $campaignIds has already been processed, we don't process any of the others?